### ISSUE TITLE
UnusedUses incorrectly reports unused imports when used in docblock intersection types 

### ISSUE THREAD

**Issue by tominventisbe** [slevomat/coding-standard#465]
This is related to #228, but unfortunately that issue was closed. If this is something that will not be supported, feel free to close this again, but the issue seemed to have been closed since the author found a workaround for his particular case and I've identified a case where this is in fact still a problem.

When writing tests with PHPUnit - among other places -, PHPStan can read intersection types:

```php
use SomeInterface;

// Incorrectly reported as unused and removed during autofixing as well, breaking PHPStan checks.
use PHPUnit\Framework\MockObject\MockObject;

/**
 * @return MockObject&SomeInterface
 */
public function getSomeInterfaceMock() { ... }
```

Unfortunately the `UnusedUses` sniff does not pick these up as it only recognizes the vertical pipe, but using it here results in different behavior - intersection types represent an "AND" relationship whereas the compound types represent an "OR" relationship.

The `UnusedUses` sniff is very useful, but throwing out PHPStan's intersection types to clean up unused imports is not a compromise I'm willing to make, so for now I'm working around this by having the sniff disabled, which is also unfortunate since there is no other sniff that does this as well as Slevomat does.

I do understand that it's not supported (yet), though, since PHPStan has only been on the rise fairly recently and is so far the only (?) tool that supports intersection types.

Thanks!

**Comment by kukulich** [comment]
It supposed to be solved with https://github.com/slevomat/coding-standard/issues/251
